src/osd/ECCommon.h

```diff
@@ -353,22 +367,37 @@ struct ECCommon {
 	for (auto &&extent: hpair.second.to_read) {
 	  returned.push_back(
 	    boost::make_tuple(
-	      extent.get<0>(),
-	      extent.get<1>(),
+	      extent.offset,
+	      extent.size,
 	      std::map<pg_shard_t, ceph::buffer::list>()));
 	}
       }
     }
     ReadOp() = delete;
     ReadOp(const ReadOp &) = delete; // due to on_complete being unique_ptr
     ReadOp(ReadOp &&) = default;
+
+    void refresh_complete(const hobject_t &hoid) {
+      std::list<
+        boost::tuple<
+          uint64_t, uint64_t, std::map<pg_shard_t, bufferlist>>> new_returned;
+      auto returned = complete[hoid].returned;
+      auto reads = to_read.find(hoid)->second.to_read;
+
+      auto r = returned.begin();
+      for (auto read : reads) {
+        new_returned.push_back(
+            boost::make_tuple(read.offset, read.size, r->get<2>()));
```

Comment:
So, the caller, ReadPipeline::send_all_remaining_reads, gets called from ECBackend::handle_sub_read_reply if for some reason we're unable to decode the desired shards from our returned data.  In that case, we need to convert to a full stripe read, but all of the reads we've completed so far are still valid as they must be full chunk reads -- it's just that we need more chunks.  Is this right so far?

It's really confusing that this is separate from the code in ReadPipeline::send_all_remaining_shards as it does not make sense out of context and "refresh" isn't a descriptive word.  It would help a lot if either:
a) this were inlined in send_all_remaining_reads with the above explanation as a comment or
b) the logic adjusting returned and to_read were in the same method, perhaps expand_read_bounds_to_stripe with a method comment explaining the above.

I think b) would be clearer.